lib/utils/patterns.coffee

```diff
@@ -155,6 +155,8 @@ exports.inferOrSelectDevice = (applicationName) ->
 		if applicationName?
 			return resin.models.device.getAllByApplication(applicationName)
 		return resin.models.device.getAll()
+	.filter (device) ->
+		resin.models.device.isOnline(device.uuid)
```

Comment:
This would cause a new HTTP request to the API for every device on the list, which is not very efficient. Notice that you already have information about the device online state in the object array returned by `resin.models.device.getAll()` or `resin.models.device.getAllByApplication()`, which I believe is an integer `is_online` property.